libazureinit/src/user.rs

```diff
@@ -50,6 +50,29 @@ pub(crate) fn provision_ssh(
     Ok(())
 }
 
+pub fn add_user_for_passwordless_sudo(username: &str) -> Result<(), Error> {
+    // Create a file under /etc/sudoers.d with azure-init-user
+    let sudoers_path = "/etc/sudoers.d/azure-init-user";
```

Comment:
We will want to write a unit test or two for this function, and as you've likely noticed we can't mock things out in the same way one can in, say, Python. Setting the file path here makes writing unit tests difficult since we probably want to write the test file out to some temporary directory.

However, we can use a trick to make this easier: rather than defining the path here, we can accept the path as a function parameter. In our unit tests we'll create a temporary directory and pass a path inside that to this function so we can make sure it creates the file we want with the permissions we expect.

This trick has the somewhat pompous (IMO) name of "dependency injection" which really just means "give the function the thing it needs rather than have it create it", but it's something you'll likely reach for a lot more in Rust than Python since you can't mock things out in the same way.

Reply:
I have created a test case and modified the function as suggested by giving path as parameter, can you take a look at let me know any changes required.